program/src/state/pool.rs

```diff
@@ -298,14 +301,20 @@ impl Pool {
                     .checked_pow(offset.into())
                 });
 
-                let result = match direction {
+                let result = unwrap_int!(match direction {
                     // price * (1 + delta)^trade_count
                     Direction::Up => base.checked_mul(&factor),
                     //same but / instead of *
                     Direction::Down => base.checked_div(&factor),
-                };
+                });
 
-                unwrap_int!(u64::try_from(unwrap_checked!({ result?.to_imprecise() })).ok())
+                // Round up for buys (Direction::Up), down for sells (Direction::Down)
+                let rounded_result = unwrap_int!(match direction {
+                    Direction::Up => result.ceiling(),
+                    Direction::Down => result.floor(),
+                });
```

Comment:
Hmmm I don't think this quite aligns with "rounding up when buying, down when selling".

Since someone could still be buying when offset < 0 and thus direction = down (eg if they're on the left side of the curve relative to the starting price on a trade pool).

Might be better to just be consistent (floor always)?

Reply:
Adjusted